### ISSUE TITLE
Anonymous class implementing interface with @Nullable type parameter fails with "cannot override"

### ISSUE THREAD

**Issue by rcchan** [typetools/checker-framework#1357]
```java
interface A<T> { T method(); }
  void test() {
    A a = new A</*@Nullable*/ Object>() {
      @Override
      public @Nullable Object method() { return null; }
    };
  }
```

```
error: [override.return.invalid] @Initialized @Nullable Object method(CLASS.@Initialized @NonNull <anonymous CLASS$1> this) in <anonymous CLASS$1> cannot override @Initialized @NonNull Object method(@Initialized @NonNull A<@Initialized @NonNull Object> this) in CLASS.A; attempting to use an incompatible return type
      public @Nullable Object method() { return null; }
                       ^
  found   : @Initialized @Nullable Object
  required: @Initialized @NonNull Object
```

It doesn't seem that /*@Nullable*/ Object, specifically the @Nullable part, is being parameterized into T.  The interface shouldn't be annotated with @Nullable as some implementations shouldn't be nullable.

**Comment by wmdietl** [comment]
I cannot reproduce the issue. Using the following complete program:

```java
import org.checkerframework.checker.nullness.qual.Nullable;

interface A<T> { T method(); }

class Demo {
  void test() {
    A a = new A<@Nullable Object>() {
      @Override
      public @Nullable Object method() { return null; }
    };
  }
}
```

A current Checker Framework version, e.g. the one at http://eisop.uwaterloo.ca/live , doesn't produce an error.

Note that `interface A<T>` implicitly means `interface A<T extends @Nullable Object`; that is, the upper bound implicitly is nullable. I'm not sure whether you mean this with your last sentence.

Can you produce a complete, self-contained example of the issue? Thanks!

**Comment by rcchan** [comment]
I can't reproduce on that site either. I'll try the simple demo again on my (likely older) copy of checker.

**Comment by rcchan** [comment]
Interestingly, it only reproduces on my version with this code (interface within class?); the one in your comment doesn't reproduce the issue (succeeds).

In any case, I can verify that this fails on my version but works on the hosted one:

```java
import org.checkerframework.checker.nullness.qual.Nullable;

class Demo {
  interface A<T> { T method(); }
  void test() {
    A a = new A</*@Nullable*/ Object>() {
      @Override
      public @Nullable Object method() { return null; }
    };
  }
}

```

Closing due to bug likely being already fixed in latest.
